wbooks/src/App/Main/BookDetail/index.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import './styles.css';
+import Details from './Details';
+import Recommendations from './Recommendations';
+import CommentsSection from './CommentsSection';
+
+class BookDetail extends React.Component {
+  render() {
+    const id = this.props.match.params.number - 1;
```

Comment:
Instead of using the `id - 1` as an index, find the book by id.
This way you are not depending of the books order.